external/kafka-0-10-sql/src/main/scala/org/apache/spark/sql/kafka010/KafkaSourceProvider.scala

```diff
@@ -83,9 +83,9 @@ private[kafka010] class KafkaSourceProvider extends DataSourceRegister
         .toMap
 
     val startingStreamOffsets =
-      caseInsensitiveParams.get(STARTING_OFFSETS_OPTION_KEY).map(_.trim.toLowerCase) match {
-        case Some("latest") => LatestOffsetRangeLimit
-        case Some("earliest") => EarliestOffsetRangeLimit
+      caseInsensitiveParams.get(STARTING_OFFSETS_OPTION_KEY).map(_.trim) match {
+        case a @ Some(offset) if offset.toLowerCase.equals("latest") => LatestOffsetRangeLimit
```

Comment:
Could you just write a utility method to get offsets from caseInsensitiveParams?